code/game/gamemodes/dynamic/dynamic_midround_rolling.dm

```diff
@@ -41,36 +41,48 @@
 
 	for (var/datum/dynamic_ruleset/midround/ruleset in midround_rules)
 		if (ruleset.weight == 0)
+			log_game("DYNAMIC: FAIL: [ruleset] has a weight of 0")
 			continue
 
 		if (!ruleset.acceptable(GLOB.alive_player_list.len, threat_level))
+			log_game("DYNAMIC: FAIL: [ruleset] is not acceptable with the current parameters. Alive players: [GLOB.alive_player_list.len], threat level: [threat_level]")
 			continue
 
 		if (mid_round_budget < ruleset.cost)
+			log_game("DYNAMIC: FAIL: [ruleset] is too expensive, and cannot be bought. Midround budget: [mid_round_budget], ruleset cost: [ruleset.cost]")
 			continue
 
 		if (ruleset.minimum_round_time > world.time - SSticker.round_start_time)
+			log_game("DYNAMIC: FAIL: [ruleset] is trying to run too early. Minimum round time: [ruleset.minimum_round_time], current round time: [world.time - SSticker.round_start_time]")
 			continue
 
 		// If admins have disabled dynamic from picking from the ghost pool
 		if(istype(ruleset, /datum/dynamic_ruleset/midround/from_ghosts) && !(GLOB.ghost_role_flags & GHOSTROLE_MIDROUND_EVENT))
+			log_game("DYNAMIC: FAIL: [ruleset] is a from_ghosts ruleset, but ghost roles are disabled")
 			continue
 
 		ruleset.trim_candidates()
-		if (ruleset.ready())
-			var/ruleset_is_heavy = (ruleset.midround_ruleset_style == MIDROUND_RULESET_STYLE_HEAVY)
-			if (ruleset_is_heavy)
-				drafted_heavies[ruleset] = ruleset.get_weight()
-			else
-				drafted_lights[ruleset] = ruleset.get_weight()
-
-	if (spawn_heavy && drafted_heavies.len > 0 && pick_midround_rule(drafted_heavies))
+		if (!ruleset.ready())
+			log_game("DYNAMIC: FAIL: [ruleset] is not ready()")
+			continue
+
+		var/ruleset_is_heavy = (ruleset.midround_ruleset_style == MIDROUND_RULESET_STYLE_HEAVY)
+		if (ruleset_is_heavy)
+			drafted_heavies[ruleset] = ruleset.get_weight()
+		else
+			drafted_lights[ruleset] = ruleset.get_weight()
+
+	var/heavy_light_log_count = "[drafted_heavies.len] heavies / [drafted_lights.len] lights"
+
+	log_game("DYNAMIC: Rolling [spawn_heavy ? "HEAVY" : "LIGHT"]... [heavy_light_log_count]")
+
+	if (spawn_heavy && drafted_heavies.len > 0 && pick_midround_rule(drafted_heavies, "heavy rulesets"))
 		return
-	else if (drafted_lights.len > 0 && pick_midround_rule(drafted_lights))
+	else if (drafted_lights.len > 0 && pick_midround_rule(drafted_lights, "light rulesets"))
 		if (spawn_heavy)
-			dynamic_log("A heavy ruleset was intended to roll, but there weren't any available.")
+			dynamic_log("A heavy ruleset was intended to roll, but there weren't any available. [heavy_light_log_count]")
```

Comment:
wouldn't this give more info as a subif inside spawn_heavy

Reply:
What do you mean by that?

Reply:
IDK honestly old me sucks